src/types/abstract_hmm.jl

```diff
@@ -23,18 +23,18 @@ Any `AbstractHMM` which satisfies the interface can be given to the following fu
 - [`forward_backward`](@ref)
 - [`baum_welch`](@ref) (if `[fit!](@ref)` is implemented)
 """
-abstract type AbstractHMM end
+abstract type AbstractHMM{ar} end
 
 @inline DensityInterface.DensityKind(::AbstractHMM) = HasDensity()
 
 ## Interface
 
 """
-    length(hmm)
+    size(hmm, control)
```

Comment:
The number of states is independent from the control value so `size` should not take a second argument. Especially since the syntax `size(x, dim)` means the length along a specific dimension.

Reply:
We can just require that the user define length for their own AbstractHMM subtype, and then all of our problems go away